### ISSUE TITLE
Last changes with hash function is not good at all

### ISSUE THREAD

**Issue by Renesanse** [felangel/equatable#43]
code: 
```
class ComparableWrapper extends Equatable {
  final Object property;
  ComparableWrapper(this.property);

  @override
  List<Object> get props => [property];
}
```

v1.0.0: 
code: 
```
    final map = {1: '1', 2: '2'};
    print(ComparableWrapper(map).hashCode);
    map.remove(1);
    print(ComparableWrapper(map).hashCode);
```
console: 
1061099485
1061099485

v0.6.1: 
code: 
```
    final map = {1: '1', 2: '2'};
    print(ComparableWrapper(map).hashCode);
    map.remove(1);
    print(ComparableWrapper(map).hashCode);
```
console: 
343974154
350953922

Well, I am understand you want to find a `good` hash function, but I think last one was better =) Thx

**Comment by felangel** [comment]
Hi @Renesanse 👋 
Thanks for opening an issue and for catching this case! I'll have a fix asap 👍

**Comment by felangel** [comment]
Fixed in #44 and published in v1.0.1

**Comment by Renesanse** [comment]
Thx
